[inductor] Remove usage of device_interface from _inductor.runtime

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #124681
* __->__ #124592



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D56723770](https://our.internmc.facebook.com/intern/diff/D56723770)

jansel committed:
Update

[ghstack-poisoned]

jansel committed:
Update

[ghstack-poisoned]

masnesral commented on torch/_inductor/runtime/triton_heuristics.py:
```diff
@@ -219,19 +217,19 @@ def precompile(self, warm_cache_only_with_cc=None):
-                # Disable for Intel GPU as Triton is not ready to return n_regs for a compiled_binary.
-                and self.device_type != "xpu"
+                # Disable for AMDGPU/Intel as Triton is not ready to return n_regs for a compiled_binary.
+                and device_prop.type == "cuda"
```
Just checking this is what you want. Instead of != xpu and != hip, you're doing == cuda. I assume that's fine?

jansel replied:
Yes, these are the only 3 supported values and if we add new ones we likely want to skip this.

masnesral commented on torch/_inductor/runtime/triton_heuristics.py:
```diff
@@ -292,15 +290,21 @@ def precompile(self, warm_cache_only_with_cc=None):
 
-    def _precompile_config(self, cfg: Config, warm_cache_only_with_cc: Optional[int]):
+    def get_device_interface(self):
+        # this code cannot run in compile workers, because it imports from torch
```
I wonder if there's a way we could assert that?

jansel replied:
There are asserts in the `verify()` function in the followup PR.

masnesral commented on torch/_inductor/runtime/triton_heuristics.py:
```diff
@@ -314,22 +318,9 @@ def _precompile_config(self, cfg: Config, warm_cache_only_with_cc: Optional[int]
-            cc = self.device_interface.get_compute_capability(device)
-
-        compile_meta["cc"] = cc
+        # device type will be "hip" rather than "cuda" here
```
nit: Not sure I understand this comment. Seems like it _could_ be hip here.

jansel replied:
yes correct

masnesral reviewed (approved):
LGTM

jansel commented:
@pytorchbot merge

jansel committed:
Update

[ghstack-poisoned]

jeanschmidt commented:
@pytorchbot revert -m "breaking internal tests, check D56522594" -c nosignal

jeanschmidt commented:
@pytorchbot revert -m "breaking internal tests, check D56522594" -c nosignal

masnesral committed:
Update on "[inductor] Remove usage of device_interface from _inductor.runtime"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

masnesral committed:
rebase past broken benchmark on "[inductor] Remove usage of device_interface from _inductor.runtime"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

Differential Revision: [D56723770](https://our.internmc.facebook.com/intern/diff/D56723770)

[ghstack-poisoned]

masnesral commented:
@masnesral has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56723770).
